### PR TITLE
JENKINS-53361 Copy files using a regular expression

### PR DESCRIPTION
**Pull request by nrayapati**
# Description

Copy files using a regular expression

See [JENKINS-53361](https://issues.jenkins-ci.org/browse/JENKINS-53361).

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/sshsteps/steps/GetStep.java [hunk 1]
```diff
@@ -106,7 +116,7 @@ public GetCallable(GetStep step, TaskListener listener, String into) {
 
       @Override
       public Object execute() {
-        return getService().get(((GetStep) getStep()).getFrom(), into);
+        return getService().get(((GetStep) getStep()).getFrom(), into, ((GetStep) getStep()).getFilterBy(), ((GetStep) getStep()).getFilterRegex());
```
[the remark this answers is not in the document]
Comment (nrayapati) [reply src/main/java/org/jenkinsci/plugins/sshsteps/steps/GetStep.java]: Updated with https://github.com/jenkinsci/ssh-steps-plugin/pull/13/commits/8203a5ef6b3d6613c1473edac961e1eb5018af30. Thank you.

File: src/main/java/org/jenkinsci/plugins/sshsteps/steps/PutStep.java [hunk 2]
```diff
@@ -98,7 +108,7 @@ public PutCallable(PutStep step, TaskListener listener, String from) {
 
       @Override
       public Object execute() {
-        return getService().put(from, ((PutStep) getStep()).getInto());
+        return getService().put(from, ((PutStep) getStep()).getInto(), ((PutStep) getStep()).getFilterBy(), ((PutStep) getStep()).getFilterRegex());
```
[the remark this answers is not in the document]
Comment (nrayapati) [reply src/main/java/org/jenkinsci/plugins/sshsteps/steps/PutStep.java]: Updated with https://github.com/jenkinsci/ssh-steps-plugin/pull/13/commits/8203a5ef6b3d6613c1473edac961e1eb5018af30. Thank you.

File: src/main/java/org/jenkinsci/plugins/sshsteps/steps/PutStep.java [hunk 3]
```diff
@@ -28,10 +28,20 @@
...
+  private String filterBy;
+
   @DataBoundConstructor
-  public PutStep(String from, String into) {
+  public PutStep(String from, String into, String filterBy, String filterRegex) {
     this.from = from;
     this.into = into;
+    this.filterRegex = filterRegex;
+    this.filterBy = filterBy;
+    if(Util.fixEmpty(filterRegex) != null && Util.fixEmpty(filterBy) == null)
```
[the remark this answers is not in the document]
Comment (nrayapati) [reply src/main/java/org/jenkinsci/plugins/sshsteps/steps/PutStep.java]: Applied formatter, which also added braces. 
https://github.com/jenkinsci/ssh-steps-plugin/pull/13/commits/8203a5ef6b3d6613c1473edac961e1eb5018af30. Thank you.

File: src/main/java/org/jenkinsci/plugins/sshsteps/steps/GetStep.java [hunk 4]
```diff
@@ -30,15 +30,25 @@
...
   private boolean override = false;
 
   @DataBoundConstructor
-  public GetStep(String from, String into) {
+  public GetStep(String from, String into, String filterBy, String filterRegex) {
     this.from = from;
     this.into = into;
+    this.filterRegex = filterRegex;
+    this.filterBy = filterBy;
+    if(Util.fixEmpty(filterRegex) != null && Util.fixEmpty(filterBy) == null)
```
[the remark this answers is not in the document]
Comment (nrayapati) [reply src/main/java/org/jenkinsci/plugins/sshsteps/steps/GetStep.java]: Applied formatter, which also added braces. 
https://github.com/jenkinsci/ssh-steps-plugin/pull/13/commits/8203a5ef6b3d6613c1473edac961e1eb5018af30. Thank you.

File: README.adoc [hunk 5]
```diff
@@ -285,6 +285,14 @@ Put a file or directory into the remote host.
 |String, *Mandatory*
 |file or directory path on the remote node.
 
+|filterBy
+|String, *Optional*, Defaults to `name`.
```
Comment (wuchenwang) [inline README.adoc]: Does it support filtering by that other than name? I'm suggesting to condense these two to one property for now: filterByName, then document it's a regex based filter.

Comment (nrayapati) [reply README.adoc]: I haven't tried all the options, but yes it does support all the params those are available on the File object, like `path`, `absolutePath`, `file`, `directory`,  `parent`, `parentFile` and so on.. 

Mostly all the params on the File object http://docs.groovy-lang.org/docs/latest/html/groovy-jdk/java/io/File.html

**Pull request merged**